platform-api/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/stack/image/StackIcon.java

```diff
@@ -0,0 +1,77 @@
+/*******************************************************************************
+ * Copyright (c) 2012-2016 Codenvy, S.A.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Codenvy, S.A. - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.che.api.workspace.server.stack.image;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Set;
+import java.util.stream.Stream;
+
+import com.google.common.base.Objects;
+
+import org.eclipse.che.api.core.ConflictException;
+
+import static java.util.stream.Collectors.toSet;
+
+public class StackIcon {
+
+    private static final Set<String> validMediaTypes = Stream.of("image/jpeg", "image/png", "image/gif", "image/svg+xml").collect(toSet());
+
+    private final String mediaType;
+    private final byte[] data;
+
+    public StackIcon(String mediaType, byte[] data) throws IOException, ConflictException {
+
+        if (data.length > 1024 * 1024) {
```

Comment:
Can you extract this value into static variable to visually understand with what we try to compare?